allennlp/modules/token_vectorizers/embedding.py

```diff
@@ -1,19 +1,29 @@
 import gzip
 import logging
 
+from overrides import overrides
 import numpy
 import torch
 
-from allennlp.data.vocabulary import Vocabulary
+from allennlp.common import Params
 from allennlp.common.checks import ConfigurationError
+from allennlp.data import Vocabulary
+from allennlp.experiments import Registry
+from allennlp.modules import TokenVectorizer
+
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
-class Embedding(torch.nn.Module):
+@Registry.register_token_vectorizer("embedding")
+class Embedding(TokenVectorizer):
     """
-    A more featureful embedding module than the default in Pytorch.
-    Adds the ability to pre-specify the weight matrix, use a non-trainable embedding and
-    catches some ill-advised use cases, such as 1 dimensional embeddings.
+    A more featureful embedding module than the default in Pytorch.  Adds the ability to
+    pre-specify the weight matrix or use a non-trainable embedding.
+
+    Note that it is not recommended that you use this directly when building models.  Instead, you
```

Comment:
This is also an attempt to explain why the `Embedding` layer is in a somewhat awkward location.

Related to that, I went back and forth between these two options: `TokenEmbedder` and `TokenVectorizer`, vs. `TextFieldEmbedder` and `TokenEmbedder` (for what is currently `TokenVectorizer`).  I ended up picking the first option because of what's currently in `NlpApi`, but that's going away, so I'm leaning towards the second option.  What do you think?  Any better naming ideas?